### PR TITLE
fix bad-data-bug with multi msgs in module evm

### LINKED ISSUE #667: There may be bad data points in CommitStateDB when exec evm transaction with multi msgs
__System info:__ [Include Ethermint commit, operating system name, and other relevant details]

* branch development

__Additional info:__ [Include gist of relevant config, logs, etc.]

* The "storage" data  from "CommitStateDB" are written to the "Store"(keeper) at the time of the "deliverTx"(handler) and are cleared at the "endBlock" stage.
* The "code" data  from "CommitStateDB" are written to the "Store"(keeper) and cleared at the "endBlock" stage.

*  If there are 2 msgs in a tx, in the deliverTx (handler) stage, where msg1 executes successfully and msg2 fails, then all store data（Keeper） will be rolled back. However, the storage data in commitStateDB and the code are still Reserved, if another tx2 is executed later in the deliverTx phase, the bad data points from tx1 will be used

* The problem encountered is Similar to [issue668](https://github.com/cosmos/ethermint/issues/668)，[issue669](https://github.com/cosmos/ethermint/issues/669)

__solution:__

1. Multiple msgs are not allowed in one tx

2. Update and clear all cache data of commitStateDB in the handler phase

### PR DESCRIPTION
**Pull request by summerpro**
Closes: #667 

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Comment by summerpro** [comment]
this pr can solve #667

File: app/ante/ante.go [hunk 1]
```diff
@@ -49,6 +49,7 @@ func NewAnteHandler(ak auth.AccountKeeper, evmKeeper EVMKeeper, sk types.SupplyK
 				authante.NewSigGasConsumeDecorator(ak, sigGasConsumer),
 				authante.NewSigVerificationDecorator(ak),
 				authante.NewIncrementSequenceDecorator(ak), // innermost AnteDecorator
+				NewValidateMsgHandlerDecorator(),
 			)
 
 		case evmtypes.MsgEthereumTx:
```
Comment (fedekunze) [inline app/ante/ante.go]: you can move this to be the top case and then remove the `MsgEthereumTx` on the ValidateMsgHandlerDecorator

Comment (summerpro) [reply app/ante/ante.go]: ok

**Comment by summerpro** [comment]
@fedekunze can this pr be merged?

**Comment by summerpro** [comment]
@fedekunze hi, this solution is indeed quite temporary. I think you may be thinking of refactoring "CommitStateDB" to solve this problem in the future. Am I right?

**Pull request closed without merging**